### PR TITLE
Retrieve and filter price list for AWS

### PR DESCRIPTION
**Pull request by kreddyj**
**What this PR does / why we need it**:
This commit defines the RateCard structure that Purser follows and
it retrieves aws pricing.

**Which issue(s) this PR fixes**:
Fixes a task in #145 

AWS pricing api for `us-east-1`: https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/us-east-1/index.json

### CODE REVIEW

File: pkg/pricing/aws/convert.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+			}
+		}
+	}
+	return nil
+}
+
+func getStorageInstancePrice(product Product, planList PlanList) *models.StoragePrice {
+	for _, pricingAttributes := range planList.OnDemand[product.Sku] {
+		for _, pricingData := range pricingAttributes.PriceDimensions {
+			for _, pricePerUnit := range pricingData.PricePerUnit {
```
Comment (hemani19) [inline pkg/pricing/aws/convert.go]: This is the same code as above. Can we abstract the common code to a separate method?

File: pkg/pricing/aws/convert.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+func getResourcePricesFromAWSPricing(awsPricing *Pricing) ([]*models.NodePrice, []*models.StoragePrice) {
+	var nodePrices []*models.NodePrice
+	var storagePrices []*models.StoragePrice
+	products := awsPricing.Products
+	planList := awsPricing.Terms
+
+	duplicateComputeInstanceChecker := make(map[string]bool)
+	for _, product := range products {
+		if product.ProductFamily == computeInstance {
+			key := product.Sku + product.Attributes.InstanceType + product.Attributes.OperatingSystem
```
Comment (hemani19) [inline pkg/pricing/aws/convert.go]: Probably create a separate method for this for better readability.

File: pkg/pricing/aws/convert.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+// GetRateCardForAWS takes region as input and returns RateCard and error if any
+func GetRateCardForAWS(region string) (*models.RateCard, error) {
+	awsPricing, err := GetAWSPricing(region)
+	if err != nil {
+		return nil, err
+	}
+	rateCard := convertAWSPricingToPurserRateCard(region, awsPricing)
+	return rateCard, nil
+}
```
Comment (hemani19) [inline pkg/pricing/aws/convert.go]: Handle error in here itself and no need to return it.

File: pkg/pricing/aws/convert.go (hunk 1 above)
Comment (kreddyj) [reply pkg/pricing/aws/convert.go]: Done

File: pkg/pricing/aws/convert.go (hunk 2 above)
Comment (kreddyj) [reply pkg/pricing/aws/convert.go]: Done

File: pkg/pricing/aws/convert.go (hunk 3 above)
Comment (kreddyj) [reply pkg/pricing/aws/convert.go]: Done

**Pull request merged**